Felienne/hedy
# [IMPROVEMENT] Overall improvements on parsons problem implementation

**Description**
In this PR we make several small improvements to the parsons problem implementation to make it more scalable and readable for the future. First of all we split the `get_trimmed_code()` function into one dedicated for "normal" code and one for parsons code. We also verify if there is a mistake in the order, if so: don't parse and only highlight the faulty lines. The prevents weird Hedy correct programs that are not in the correct order of lines as well as difficult to understand error messages.

Lastly we implement another important feature: logging of parsons attempts. Each line order attempt is logged in combination with the level, timestamp and username (or session id). Enabling us to perform some useful data analysis later on.

**Fixes**
This PR fixes #2786.

**How to test**
Verify the following improvements to make sure they are all working:
- When a faulty order is given the program is not parsed
- However, the attempt is logged into the database
- The stop button is correctly reversed into a run button when getting an error
